java-checks/src/main/java/org/sonar/java/checks/CatchUsesExceptionWithContextCheck.java

```diff
@@ -78,11 +83,42 @@ public void scanFile(JavaFileScannerContext context) {
     if (context.getSemanticModel() != null) {
       scan(context.getTree());
     }
+    excludedCatchTrees.clear();
+  }
+
+  @Override
+  public void visitTryStatement(TryStatementTree tree) {
+    if (containsEnumValueOf(tree.block())) {
+      tree.catches().stream()
+        .filter(c -> c.parameter().symbol().type().is("java.lang.IllegalArgumentException"))
+        .findAny()
+        .ifPresent(excludedCatchTrees::add);
+    }
+    super.visitTryStatement(tree);
+  }
+
+  private static boolean containsEnumValueOf(Tree tree) {
+    EnumValueOfVisitor visitor = new EnumValueOfVisitor();
+    tree.accept(visitor);
+    return visitor.hasEnumValueOf;
+  }
+
+  private static class EnumValueOfVisitor extends BaseTreeVisitor {
+    private static final MethodMatcher ENUM_VALUE_OF = MethodMatcher.create().typeDefinition("java.lang.Enum").name("valueOf").withAnyParameters();
```

Comment:
Matcher should also match subtypes of enum otherwise we are not going to catch usages like `MyEnum.valueOf("C");`